internal/commands/vaultsecrets/secrets/create.go

```diff
@@ -423,20 +364,56 @@ func readPlainTextSecret(opts *CreateOpts) error {
 	return nil
 }
 
-func readConfigFile(opts *CreateOpts) (SecretConfig, error) {
-	var sc SecretConfig
+func readConfigFile(opts *CreateOpts) (SecretConfig, detailsInternal, error) {
+	var (
+		sc SecretConfig
+		di detailsInternal
+	)
 
-	f, err := os.ReadFile(opts.SecretFilePath)
-	if err != nil {
-		return sc, fmt.Errorf("unable to open config file: %w", err)
+	if err := hclsimple.DecodeFile(opts.SecretFilePath, nil, &sc); err != nil {
+		return sc, di, fmt.Errorf("failed to decode config file: %w", err)
 	}
 
-	err = yaml.Unmarshal(f, &sc)
+	detailsMap, err := ctyValueToMap(sc.Details)
 	if err != nil {
-		return sc, fmt.Errorf("unable to unmarshal config file: %w", err)
+		return sc, di, err
+	}
+	di.Details = detailsMap
+
+	return sc, di, nil
+}
+
+func ctyValueToMap(value cty.Value) (map[string]any, error) {
+	varMapNow := make(map[string]any)
```

Comment:
Nit: this map's name is a bit confusing (with `Now` suffix)

Reply:
Good catch - leftover from manual testing